addons/spreadsheet/models/spreadsheet_mixin.py

```diff
@@ -28,13 +28,13 @@ class SpreadsheetMixin(models.AbstractModel):
 
     @api.constrains("spreadsheet_binary_data")
     def _check_spreadsheet_data(self):
-        if not(tools.config['test_enable'] or tools.config['test_file']):
-            return None
         for spreadsheet in self.filtered("spreadsheet_binary_data"):
             try:
                 data = json.loads(base64.b64decode(spreadsheet.spreadsheet_binary_data).decode())
             except (json.JSONDecodeError, UnicodeDecodeError):
                 raise ValidationError(_("Uh-oh! Looks like the spreadsheet file contains invalid data."))
+            if not(tools.config['test_enable'] or tools.config['test_file']):
+                return None
```

Comment:
otherwise it only checks the first spreadsheet in `self`
```suggestion
                continue
```